ragna/deploy/_api.py

```diff
@@ -1,107 +1,56 @@
-            )
-            document.metadata = metadata
-            _database.add_document(
-                session, user=user, document=document, metadata=metadata
-            )
-            return schemas.DocumentUpload(parameters=parameters, document=document)
-
-    # TODO: Add UI support and documentation for this endpoint (#406)
     @router.post("/documents")
-    async def create_documents_upload_info(
-        user: UserDependency,
-        names: Annotated[list[str], Body(..., embed=True)],
-    ) -> list[schemas.DocumentUpload]:
-        with _database.get_session() as session:
-            document_metadata_collection = []
-            document_upload_collection = []
-            for name in names:
-                document = schemas.Document(name=name)
-                metadata, parameters = await config.document.get_upload_info(
-                    config=config, user=user, id=document.id, name=document.name
-                )
-                document.metadata = metadata
-                document_metadata_collection.append((document, metadata))
-                document_upload_collection.append(
-                    schemas.DocumentUpload(parameters=parameters, document=document)
-                )
-
-            _database.add_documents(
-                session,
-                user=user,
-                document_metadata_collection=document_metadata_collection,
-            )
-            return document_upload_collection
-
-    # TODO: Add new endpoint for batch uploading documents (#407)
-    @router.put("/document")
-    async def upload_document(
-        token: Annotated[str, Form()], file: UploadFile
-    ) -> schemas.Document:
-        if not issubclass(config.document, ragna.core.LocalDocument):
-            raise HTTPException(
-                status_code=400,
-                detail="Ragna configuration does not support local upload",
-            )
-        with _database.get_session() as session:
-            user, id = ragna.core.LocalDocument.decode_upload_token(token)
-            document = _database.get_document(session, user=user, id=id)
-
-            core_document = cast(
-                ragna.core.LocalDocument, engine._to_core.document(document)
-            )
-            core_document.path.parent.mkdir(parents=True, exist_ok=True)
-            async with aiofiles.open(core_document.path, "wb") as document_file:
-                while content := await file.read(1024):
-                    await document_file.write(content)
-
-            return document
+    def register_documents(
+        user: UserDependency, document_registrations: list[schemas.DocumentRegistration]
+    ) -> list[schemas.Document]:
+        return engine.register_documents(
+            user=user, document_registrations=document_registrations
+        )
+
+    @router.put("/documents")
+    async def upload_documents(
+        user: UserDependency, documents: list[UploadFile]
```

Comment:
It does not. The [`UploadFile`](https://fastapi.tiangolo.com/tutorial/request-files/#file-parameters-with-uploadfile) from FastAPI actually wraps a [`SpooledTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile), i.e. it will only be kept in memory for small files and otherwise temporarily be stored on disk.